src/client/consumer.rs

```diff
@@ -449,13 +449,11 @@ mod tests {
         assert_eq!(record_and_offset.offset, 3);
         assert_eq!(high_watermark, 5);
 
-        let (record_and_offset, high_watermark) =
-            stream.next().now_or_never().unwrap().unwrap().unwrap();
+        let (record_and_offset, high_watermark) = stream.next().await.unwrap().unwrap();
```

Comment:
Perhaps these could be timeout with very small timeout?

Reply:
done